### PR TITLE
WIP: attempt to determine why JPEG2000 support isn't working

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Do not merge. Attempt to debug #227 - `configure` works fine on my machine so something is different on the CI's. Also try to force the python tests to happen.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,12 +17,13 @@ source:
...
 
 build:
-  number: 3
+  number: 4
   skip: True  # [win and vc<14]
 
 requirements:
   build:
     - cmake  # [win]
+    - pkg-config  # [unix]
```
Comment (gillins) [inline recipe/meta.yaml]: I'm now getting `OpenJPEG support:          yes` on the CI's with this which is a good start...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -186,6 +188,11 @@ outputs:
         - osgeo._gdalconst
         - osgeo._ogr
         - osgeo._osr
+      commands:
+        - $PYTHON run_tests.py  # [unix]
+        - %PYTHON% run_tests.py  # [win]
+        - $PYTHON extra_tests.py  # [unix]
+        - %PYTHON% extra_tests.py  # [win]
       about:
```
Comment (gillins) [inline recipe/meta.yaml]: ... but I still can't work out how to get these tests to run. Any ideas?
```
can't open file 'run_tests.py': [Errno 2] No such file or directory
```

Comment (kmuehlbauer) [reply recipe/meta.yaml]: @gillins It's `run_test.py` not `run_tests.py`

Comment (gillins) [reply recipe/meta.yaml]: :man_facepalming: well spotted!

Comment (kmuehlbauer) [reply recipe/meta.yaml]: @gillins Looking at `run_test.py`, `extra_tests.py` is imported there. So we do not need to call it specifically, or do I miss something?

Comment (kmuehlbauer) [reply recipe/meta.yaml]: Not yet, still thinking...

File: recipe/meta.yaml (hunk 1 above)
Comment (gillins) [reply recipe/meta.yaml]: Not sure if I need to add `run_test.sh` to the test section here also to get it to run...

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

File: recipe/install_python.bat [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+%PYTHON% setup.py build_py
+if errorlevel 1 exit 1
+
+%PYTHON% setup.py build_scripts
+if errorlevel 1 exit 1
+
+%PYTHON% setup.py install
+if errorlevel 1 exit 1
+
```
Comment (ocefpaf) [inline recipe/install_python.bat]: I guess we also need to remove `bld.bat`, right?

BTW, this is not how they did in AR, see https://github.com/AnacondaRecipes/gdal-feedstock/tree/master/recipe

Not sure if there is a reason or just style preference. Ping @msarahan here to comment on it.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -143,6 +144,8 @@ outputs:
     test:
       files:
         - test_data
+        - run_test_lib.sh  # [unix]
+        - run_test_lib.bat  # [win]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: My understanding is that the `run_test` would be run automatically without the need to list it. But not sure if they are run twice or only once with both packages installed. Either way I don't think we need this change. @msarahan probably knows more.

Comment (gillins) [reply recipe/meta.yaml]: Hmm it doesn't seem to be here (or in #229). Well I can't find the outputs of these tests in the log (https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/run_test.sh#L7 and https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/run_test.py#L13 should print something) though I'm sure I'm missing something.

BTW if these tests were run we would have caught the missing driver as `run_test.py` checks for it...

Comment (ocefpaf) [reply recipe/meta.yaml]: Yep. You are, obviously, correct :grimacing: 

@msarahan I guess that it is a bug in conda-build, no? What is the recommend way to run tests for split packages in an integrated way? Or separated?

@gillins meanwhile let's merge #229 but let's keep working on this one to get the tests back up.

**Comment by gillins** [comment]
ping @msarahan. Did this ever get resolved? I think there was some confusion about how/when tests are run for split packages....

**Pull request closed without merging**